hnn_core/pyramidal.py

```diff
@@ -73,6 +73,9 @@ def get_sectnames(self):
 
     def create_dends(self, p_dend_props):
         """Create dendrites."""
+        # XXX: name should be unique even across cell types?
+        # otherwise Neuron cannot disambiguate, hence
+        # self.name + '_' + key
         for key in p_dend_props:
             self.dends[key] = h.Section(
                 name=self.name + '_' + key)  # create dend
```

Comment:
Perhaps this behaviour reflects how connections are made in [`parconnect_from_src()`](https://github.com/jonescompneurolab/hnn-core/blob/c299d31c15176439e3dccf648729d16dbfc3f4d4/hnn_core/cell.py#L279)? It could be the case that neuron relies on some concatenation of object names under the hood. 

It looks like [`syn_create()`](https://github.com/jonescompneurolab/hnn-core/blob/c299d31c15176439e3dccf648729d16dbfc3f4d4/hnn_core/cell.py#L239) is called from [`_synapse_create()`](https://github.com/jonescompneurolab/hnn-core/blob/c299d31c15176439e3dccf648729d16dbfc3f4d4/hnn_core/pyramidal.py#L200) using the dendrite objects themselves. The synapse object is also passed directly to `parconnect_from_src()` which finally calls `_PC.gid_connect()`. 

Nothing in the neuron documentation suggests that the names are being used instead of object identifiers, however the docstring on `parconnect_from_src()` indicates the argument `post_syn` (the actual synapse object being passed) is a `str`. In summary I'm still confused. But this is good to be aware when we dig into an API for connectivity.

Reply:
humm ... okay 2 issues here:

1) Documentation of `parconnect_from_src` is wrong. Perhaps an easy fix for someone new? It should be an instance of `Exp2Syn` not `str` @kohl-carmen or @samikane ?

2) Neuron Python objects are really wrappers around hoc stuff. This issue reminds me of how connections are made in Neuron. You need to do it in 2 steps: a) create a [NetCon object](https://www.neuron.yale.edu/neuron/static/py_doc/modelspec/programmatic/network/netcon.html#NetCon), and b) assign the NetCon object a gid (using [gid_connect](https://www.neuron.yale.edu/neuron/static/py_doc/modelspec/programmatic/network/parcon.html#ParallelContext.gid_connect)). If they were Python objects, I would have expected this to happen in one step

conclusion -- I don't understand how these things work either. Just discovering things as I change the code ;)

Reply:
Based off of the NEURON [hoc documentation](https://www.neuron.yale.edu/neuron/static/new_doc/modelspec/programmatic/topology/secspec.html), it looks like calling `h.Section(name='fun_seg')` would actually create a hoc object accessible by the name `fun_seg` from a hypothetical hoc interpreter in the appropriate namespace. I'm guessing gids, however, work by appending hoc objects into a vector of some sort that are indexed by gid.